lib/gradebook.rb

```diff
@@ -1 +1,29 @@
+class GradeBook
+  attr_reader :instructor, :courses
+  
+  def initialize(instructor)
+    @instructor = instructor
+    @courses = []
+  end
 
+  def add_course(course)
+    @courses << course
+  end
+
+  def list_all_students
+    lists = {}
+    lists = @courses.map do |course|
+      course.students
+    end.flatten
+  end
```

Comment:
the first `lists` variable is being overwritten by the second. This method is returning an array of all students, instead of the requested hash.
```suggestion
  def list_all_students
    list = Hash.new([]) # this syntax sets a default value of an empty array for every value in the new hash
    @courses.each do |course|
      list[course] = students
    end
    list
  end
```